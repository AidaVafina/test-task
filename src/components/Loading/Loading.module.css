.loadingContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 2rem;
}

.spinner {
    width: 40px;
    height: 40px;
    border: 4px solid var(#ddd);
    border-top: 4px solid var(--color-primary, #3498db);
    border-radius: 50%;
    animation: spin 1s linear infinite;
    margin-bottom: 0.5rem;
}

@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}
