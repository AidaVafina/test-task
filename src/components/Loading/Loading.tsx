import React from 'react';
import styles from './Loading.module.css';

const Loading: React.FC = () => {
    return (
        <div className={styles.loadingContainer}>
            <div className={styles.spinner} />
            <p>Загрузка...</p>
        </div>
    );
};

export default Loading;
