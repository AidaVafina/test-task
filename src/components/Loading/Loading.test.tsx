import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Loading from './Loading';
import styles from './Loading.module.css';

describe('Loading Component', () => {
    it('should match snapshot', () => {
        const { container } = render(<Loading />);
        expect(container).toMatchSnapshot();
    });

    it('should render loading text', () => {
        const { container } = render(<Loading />);
        expect(container.textContent).toContain('Загрузка...');
    });

    it('should render spinner', () => {
        const { container } = render(<Loading />);
        expect(container.querySelector(`.${styles.spinner}`)).toBeInTheDocument();
    });
}); 