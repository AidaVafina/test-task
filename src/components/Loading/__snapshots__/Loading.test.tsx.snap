// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading Component > should match snapshot 1`] = `
<div>
  <div
    class="_loadingContainer_0d12fb"
  >
    <div
      class="_spinner_0d12fb"
    />
    <p>
      Загрузка...
    </p>
  </div>
</div>
`;

exports[`Loading Component should match snapshot 1`] = `
<div>
  <div
    class="loadingContainer"
  >
    <div
      class="spinner"
    />
    <p>
      Загрузка...
    </p>
  </div>
</div>
`;
